hammr/commands/image/image.py

```diff
@@ -707,31 +668,48 @@ def get_pimage_from_id(self, id):
             return 2
         return pimage
 
-    def get_deployment_from_args_for_deploy(self, args):
-        deployment = Deployment()
-        myinstance = Instance()
+    def deploy_aws(self, file, pimage):
+        image_id = generics_utils.extract_id(pimage.imageUri)
+        deployment = build_deployment_amazon(file)
+        if deployment is None:
+            return
 
-        if args.deploy_name:
-            deployment.name = args.deploy_name
-        else:
-            printer.out("No name given for the deployment", printer.ERROR)
-            return ""
-        if args.vcpu:
-            myinstance.cores = str(args.vcpu)
-        else:
-            myinstance.cores = "1"
-        if args.memory:
-            myinstance.memory = str(args.memory)
-        else:
-            myinstance.memory = "1024"
-        deployment.instances = pyxb.BIND()
-        deployment.instances._ExpandedName = pyxb.namespace.ExpandedName(Namespace, 'Instances')
-        deployment.instances.append(myinstance)
-
-        return deployment
-
-    def is_targeted_cloud_amazon(self, pimage):
-        if pimage.targetFormat:
-            target_platform = pimage.targetFormat.name
-            return target_platform == "Amazon AWS"
-        return False
+        deployed_instance = call_deploy(self, pimage, deployment, image_id)
+        if deployed_instance == 2:
+            return 2
+        deployed_instance_id = deployed_instance.applicationId
+
+        print("Deployment in progress")
+
+        status = show_deploy_progress_aws(self, deployed_instance_id)
+        return print_deploy_info(self, status, deployed_instance_id)
+
+    def deploy_openstack(self, file, pimage):
+        image_id = generics_utils.extract_id(pimage.imageUri)
+        pid = pimage.dbId
+        bar_status = OpStatus()
+        progress = create_progress_bar_openstack(bar_status)
+        deployment = build_deployment_openstack(file, pimage, pid, retrieve_credaccount(self, pid, pimage))
+        if deployment is None:
+            return
+
+        bar_status.percentage = 50
+        bar_status.message = "Deploying instance"
+        progress.update(bar_status.percentage)
+
+        deployed_instance = call_deploy(self, pimage, deployment, image_id)
+        if deployed_instance == 2:
+            return 2
+        deployed_instance_id = deployed_instance.applicationId
```

Comment:
The AWS deploy method and the openstack one are not symmetric.
One call use a submethod "show_deploy_progress_aws", one print "Deployment in progress".
I think it will improve readability to have them as similar as possible.